Doc/library/atexit.rst

```diff
@@ -14,11 +14,11 @@ functions.  Functions thus registered are automatically executed upon normal
 interpreter termination.  :mod:`atexit` runs these functions in the *reverse*
 order in which they were registered; if you register ``A``, ``B``, and ``C``,
 at interpreter termination time they will be run in the order ``C``, ``B``,
-``A``.
-
-**Note:** The functions registered via this module are not called when the
+``A``.  The functions registered via this module are not called when the
 program is killed by a signal not handled by Python, when a Python fatal
-internal error is detected, or when :func:`os._exit` is called.
+internal error is detected, or when :func:`os._exit` is called.  Equality
+comparisons are used internally during registration and unregistration
```

Comment:
Check the code, but I think registration is just append to a list, so where would only be comparisons in unregister.

Reply:
That is correct (https://github.com/python/cpython/blob/bb3e0c240bc60fe08d332ff5955d54197f79751c/Modules/atexitmodule.c#L134-L178)

I fixed the docs accordingly.